### PR TITLE
do not need to check if element in dict input is Tensor.

### LINKED ISSUE #97229: Jit trace failed with dict inputs
### 🐛 Describe the bug

Background: I want to trace the model for generating tasks, and my input is as follows:
input_dict = {
'input_ids': Tensor,
'past_key_values': Tuple,
'attention_mask': Tensor,
'position_ids': Tensor
}
model = [gpt2](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt2/modeling_gpt2.py#L955)

when I run `torch.jit.trace(model, example_kwarg_inputs=input_dict , strict=False)`
I got this error:
image

The error is occurred on [mod(**input)](https://github.com/pytorch/pytorch/blob/master/torch/jit/_trace.py#L459)

Could you please help me check this error? Thanks!

BTW, I tried to use jit trace by tuple inputs with past_key_values and it works. I also removed past_key_values in my input dict and it also works, so the bug may have happened when the input dict contains tuples.

### Versions

Collecting environment information...
PyTorch version: 2.0.0.dev20230228+cpu-cxx11-abi
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: CentOS Stream 8 (x86_64)
GCC version: (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18)
Clang version: Could not collect
CMake version: version 3.20.2
Libc version: glibc-2.28

Python version: 3.9.16 (main, Mar  1 2023, 18:22:10)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.16.0-rc8-intel-next-01534-g53cb5f883cf7-x86_64-with-glibc2.28
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              224
On-line CPU(s) list: 0-223
Thread(s) per core:  2
Core(s) per socket:  56
Socket(s):           2
NUMA node(s):        2
Vendor ID:           GenuineIntel
CPU family:          6
Model:               143
Model name:          Genuine Intel(R) CPU 0000%@
Stepping:            3
CPU MHz:             1900.000
CPU max MHz:         1900.0000
CPU min MHz:         800.0000
BogoMIPS:            3800.00
Virtualization:      VT-x
L1d cache:           48K
L1i cache:           32K
L2 cache:            2048K
L3 cache:            107520K
NUMA node0 CPU(s):   0-55,112-167
NUMA node1 CPU(s):   56-111,168-223
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req hfi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm uintr avx512_vp2intersect md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] intel-extension-for-pytorch==2.0.0+git2cb04e0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.2
[pip3] torch==2.0.0.dev20230228+cpu.cxx11.abi
[conda] intel-extension-for-pytorch 2.0.0+git2cb04e0          pypi_0    pypi
[conda] mkl-include               2023.0.0                 pypi_0    pypi
[conda] mkl-static                2023.0.0                 pypi_0    pypi
[conda] numpy                     1.24.2                   pypi_0    pypi
[conda] torch                     2.0.0.dev20230228+cpu.cxx11.abi          pypi_0    pypi

### PR DESCRIPTION
**Pull request by sywangyi**
sometimes it's a tuple with tensor element such as past value key in text generation case

Fixes https://github.com/pytorch/pytorch/issues/97229

### CODE REVIEW

**Comment by sywangyi** [comment]
@davidberard98 @jgong5 @yao-matrix. please help review

**Review by davidberard98 — COMMENTED** [review]
thanks for the fix! could you add a test for the case that used to fail so I have a better understanding of what the issue was?  you could put it in `test/jit/test_tracer.py`

**Review by jgong5 — APPROVED** [review]
Please add the test case as @davidberard98 suggested. You may construct a simple test from the failure you tried to address at https://github.com/pytorch/pytorch/issues/97229

**Comment by sywangyi** [comment]
test case added

File: test/jit/test_tracer.py [hunk 1]
```diff
@@ -1973,6 +1973,18 @@ def forward(self, x, y, **deprecated_arguments):
...
+    def test_trace_with_tuple_tensor(self):
+        class MyClass(torch.nn.Module):
+            def __init__(self):
+                super(MyClass, self).__init__()
+
+            def forward(self, x, y):
+                return x + y[0] + y[1]
+
+        model = MyClass()
+        m2 = torch.jit.trace(model, (torch.ones(1), (torch.ones(1),torch.ones(1))))
```
Comment (jgong5) [inline test/jit/test_tracer.py:1985]: Please check the running result too.

File: test/jit/test_tracer.py [hunk 2]
```diff
@@ -1973,6 +1973,18 @@ def forward(self, x, y, **deprecated_arguments):
...
+        m3 = torch.jit.trace(model, example_kwarg_inputs={'x': torch.ones(1), "y": (torch.ones(1),torch.ones(1))}, strict=False)
```
Comment (jgong5) [inline test/jit/test_tracer.py:1986]: why `strict=False` is needed?

**Review by davidberard98 — APPROVED** [review]
LGTM! Thanks for the test!

**Pull request closed without merging**